### PR TITLE
Trial tide on operator repository

### LINKED ISSUE #1520: Enable tide on all repos
/assign @fejta @geeknoid 

@sdake @howardjohn 

Basic intent is to create CODEOWNERS files in all repos and then configure tide to search for `review:approved` issues, which is something Martin is working on and will canary on istio.io or operator.

Right now tide is configured to look for lgtm, approved labels instead, which work with OWNERS files. If a repo has these OWNERS files (istio/istio, istio/test-infra do, for example), then you can turn on tide now and auto-migrate later with the following:

* Ensure the `lgtm` and `approve` plugin are turned on at either the org (`istio`) or repo (`istio/istio`) level
  * org - https://github.com/istio/test-infra/blob/4b57822c8150b91ed404adbdd16714c6850d9313/prow/plugins.yaml#L63-L67
  * repo https://github.com/istio/test-infra/blob/4b57822c8150b91ed404adbdd16714c6850d9313/prow/plugins.yaml#L79-L85
* Configure the `lgtm` plugin to treat a github approve like an `/lgtm` comment
  * This is optional but necessary in order to have github approve add the lgtm label, otherwise you need an `/lgtm` comment.
  * https://github.com/istio/test-infra/blob/4b57822c8150b91ed404adbdd16714c6850d9313/prow/plugins.yaml#L37-L50
* Configure `tide` to respond to the repo:
  * https://github.com/istio/test-infra/blob/4b57822c8150b91ed404adbdd16714c6850d9313/prow/config.yaml#L191-L201
* Disable mergify on the repo

Repos configured this way can start using tide now and will get auto-upgraded to use CODEOWNERS instead of OWNERS.

Alternatively continue using mergify and just wait for us to validate the new tide pattern and turn this on for all repo.

### PR DESCRIPTION
**Pull request by sdake**
Fixes: #1520

### CODE REVIEW

**Comment by sdake** [comment]
/assign @fejta 

can you take a look if this is what you had meant?

If so please contact me on slack prior to merge so I can delete mergify out of operator repo.

Cheers
-steve

**Comment by sdake** [comment]
I don't think we need approve as per discussion in issue tracker:
```
{"component":"checkconfig","file":"prow/cmd/checkconfig/main.go:71","func":"main.reportWarning","level":"warning","msg":"the following orgs or repos enable at least one plugin that uses OWNERS files (approve, blunderbuss, owners-label) but do not enable the verify-owners plugin to ensure validity of OWNERS files: [repo: istio/operator]","time":"2019-07-22T23:38:21Z"}
{"component":"checkconfig","file":"prow/cmd/checkconfig/main.go:74","func":"main.reportWarning","level":"fatal","msg":"Strict is set and there were warnings","time":"2019-07-22T23:38:21Z"}
```

File: prow/plugins.yaml
Comment (howardjohn) [inline prow/plugins.yaml]: Will `/approve` work without a OWNERS file? I am not sure. An alternative would be to add a new tide query with `review: approved` as a part instead of looking for lgtm and approved

Comment (sdake) [reply prow/plugins.yaml]: I doubt it.  do we need an approved label for tide to merge the work?  The way I parsed the issue, lgtm is sufficient.

**Comment by sdake** [comment]
looks like prow or tide require the approved label:
```
{"component":"checkconfig","file":"prow/cmd/checkconfig/main.go:71","func":"main.reportWarning","level":"warning","msg":"the following orgs or repos require the approved label for merging but do not enable the approve plugin: [repo: istio/operator]","time":"2019-07-22T23:44:20Z"}
{"component":"checkconfig","file":"prow/cmd/checkconfig/main.go:74","func":"main.reportWarning","level":"fatal","msg":"Strict is set and there were warnings","time":"2019-07-22T23:44:20Z"}
```

File: prow/plugins.yaml
Comment (sdake) [reply prow/plugins.yaml]: @howardjohn your approach may work better than the proposal in the issue tracker.  What I've tried thus far is not passing the linting...

File: prow/plugins.yaml
Comment (fejta) [inline prow/plugins.yaml]: The standard tide block looks for both an approved and lgtm label (see L200 in config.yaml). So this repo will need OWNERS files and the approve plugin enabled.

If you're not wanting to do that then this repo will need its own section in tide. Make it stop looking for an approved label. If you want to make sure it gets reviewed by CODEOWNERS then you can set `reviewApprovedRequired: true` per 
 https://github.com/kubernetes/test-infra/blob/3f92da8679e28b626a91cf5f168ceee014d038f1/prow/config/tide.go#L181

**Review by fejta — COMMENTED** [review]
edit: need to resolve the approval part still actually

File: prow/plugins.yaml
Comment (fejta) [reply prow/plugins.yaml]: https://github.com/istio/test-infra/blob/c01c3a1f196fe7901aa76e1b3d9cf68a5f482883/prow/config.yaml#L193-L206 says that for the four listed repo (and this PR adds a fifth) tide will try and merge any PR that has `approved`, `lgtm` and `cla: yes` labels (avoiding any of thse PRs that have a `needs-rebase`, `do-not-merge`, etc label)

You can create another section with your own config (note that `istio-releases/pipeline` does this already). Or you can add an OWNERS file in operator and turn on the approve plugin to make sure we get the label added.

File: prow/plugins.yaml
Comment (fejta) [reply prow/plugins.yaml]: But note that essentially this is what Martin is planning to try out in the istio.io repo

Comment (sdake) [reply prow/plugins.yaml]: Ya, we have already transitioned to CODEOWNERS and off OWNERS.  I'll take a look at the link you posted and submit a followup.

Thanks!

File: prow/plugins.yaml
Comment (geeknoid) [reply prow/plugins.yaml]: The pattern we want for the operator repo is:

- CODEOWNERS and not OWNERS
- No /foo commands
- No "approve" or "lgtm" labels defined or used in the repo.
- Users approve PRs using the normal GitHub approval/comment UI.

Comment (sdake) [reply prow/plugins.yaml]: I understand @geeknoid :)

File: prow/plugins.yaml
Comment (sdake) [reply prow/plugins.yaml]: @fejta ya, @geeknoid asked me to trial this in the oeprator repo instead, as it might disrupt the automated backports that are implemented in mergify and widely used in the istio.io repo.

**Comment by sdake** [comment]
@fejta I believe this PR still needs cla: yes support implemented and that work happens in prow/config.yaml?

File: prow/config.yaml
Comment (fejta) [inline prow/config.yaml:215]: YAML is hard to read, but this is redefining the tide key.

What we want is to add this `queries` list element to the `queries` key on L192, like L193 and L207 both do. (so insert it between L214 and L215).

Also do you want to choose a squash method? If so add it to L216-L217 `istio/operator: squash`

File: prow/config.yaml
Comment (fejta) [inline prow/config.yaml]: You don't want to search for a label named `review:approved` you want to ensure github considers the PR approved, which is `reviewApprovedRequired: true` (aka a sibling to labels)

```yaml
- repos:
  - istio/operator
  labels:
  - "cla: yes"  # do we care about this label?
  missingLabels: # what you have now
  reviewApprovedRequired: true
```

File: prow/config.yaml
Comment (sdake) [reply prow/config.yaml:215]: right - will address

thanks

File: prow/config.yaml
Comment (fejta) [reply prow/config.yaml]: And since I know you stay up at night wondering about how to configure tide and prow, here's what you have to work with right now: https://github.com/kubernetes/test-infra/blob/3f92da8679e28b626a91cf5f168ceee014d038f1/prow/config/tide.go#L168-L182

**Review by fejta — COMMENTED** [review]
/hold

Please `/hold cancel` at your leisure. Tide should merge immediatelyish after.

**Review by fejta — APPROVED** [review]

File: prow/config.yaml
Comment (fejta) [inline prow/config.yaml]: `"cla: yes"` Otherwise yaml thinks you're doing `[{"cla": "yes"}]` rather than `["cla: yes"]`

File: prow/config.yaml
Comment (fejta) [inline prow/config.yaml]: This still doesn't match how it's done above. Just copy how we do it there (this also won't work, we want a label item, not a cla object)

**Comment by sdake** [comment]
mergify is gone from operator repo here: https://github.com/istio/operator/pull/108

**Review by howardjohn — APPROVED** [review]

**Pull request merged**
